### PR TITLE
Pull in contributors data at build time

### PR DESCRIPTION
**Pull request by olizilla**
- Use hugo's getJSON function to grab contributors data at build time.
- Simplify contributors impl

### CODE REVIEW

**Comment by alanshaw** [comment]
👓

File: layouts/partials/contributors.html [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+    opacity: 1;
+    position: relative;
+    z-index: 0;
+  }
+  #contributors .hex.show:hover {
+    transform: scale(1.2);
+    z-index: 10;
+  }
+  #contributors .hex.show:hover + .hex {
+  }
```
Comment (alanshaw) [inline layouts/partials/contributors.html]: can remove?

File: layouts/partials/contributors.html [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+      link.title = item.username
+      link.style.width = hexWidth + 'px'
+      link.style.height = hexHeight + 'px'
+
+      var img = new Image()
+      img.onload = function () {
+        link.style.background = '#333 url(' + item.photo + ') no-repeat center'
+        link.style.backgroundSize = 'cover'
+        cb(null, link)
+      }
```
Comment (alanshaw) [inline layouts/partials/contributors.html:142]: If the image fails to load you halt the animation I think. You need an `onerror` handler to call the callback in this case.

File: layouts/partials/contributors.html [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+          var now = Date.now()
+          var diff = now - lastRender
+          var delay = loadingDelayMs - diff
+          if (delay < 0) delay = 0
+          setTimeout(function () {
+            renderWithPause(itemIndex + 1, now)
+          }, delay)
+        })
+      }
+      renderWithPause(0, Date.now())
```
Comment (alanshaw) [inline layouts/partials/contributors.html]: Feels like you could extract this control flow logic from the rendering and publish as a module ;)

**Review by alanshaw — CHANGES_REQUESTED** [review]
:thumbsup: looking good and working great

**Comment by olizilla** [comment]
Fixed up!

File: layouts/partials/contributors.html [hunk 4]
```diff
@@ -103,21 +101,24 @@
         }
       })
       var nextRow = 0
-      function renderWithPause(itemIndex, lastRender) {
+      var lastRender = 0
+      function renderWithPause(itemIndex) {
+        if (itemIndex >= data.length) return
         renderContributor(data[itemIndex], function (err, el) {
+          var nextIndex = itemIndex + 1
+          if (err) renderWithPause(nextIndex)
```
Comment (alanshaw) [inline layouts/partials/contributors.html]: Should this `return renderWIthPause(/* ... */)` here?

File: layouts/partials/contributors.html [hunk 5]
```diff
@@ -103,21 +101,24 @@
...
-          nextRow = (nextRow + 1) % rows.length
           var now = Date.now()
           var diff = now - lastRender
           var delay = loadingDelayMs - diff
           if (delay < 0) delay = 0
+          lastRender = now
+          nextRow = (nextRow + 1) % rows.length
           setTimeout(function () {
-            renderWithPause(itemIndex + 1, now)
+            renderWithPause(itemIndex + 1)
```
Comment (alanshaw) [inline layouts/partials/contributors.html]: use `nextIndex`?

**Pull request merged**